### PR TITLE
Fixes issue #871: Automatic-Module-Names are added

### PR DESCRIPTION
**Pull request by solomax**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -66,6 +66,7 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <automatic-module-name>--module-name-need-to-be-overriden--</automatic-module-name>
```
Comment (fstab) [inline pom.xml:69]: All other property names use dots, so let's rename this to `automatic.module.name`.

File: prometheus-metrics-exposition-formats/pom.xml [hunk 2]
```diff
@@ -32,6 +32,10 @@
         </developer>
     </developers>
 
+    <properties>
+        <automatic-module-name>io.prometheus.metrics.exposition.formats</automatic-module-name>
```
Comment (fstab) [inline prometheus-metrics-exposition-formats/pom.xml:36]: The package name is `io.prometheus.metrics.expositionformats`.

**Review by fstab — CHANGES_REQUESTED** [review]
Thanks a lot @solomax!

I added a few minor comments. One more thing: We are trying to keep a similar structure across all `pom.xml` files. Please move the `<properties>` sections so that it's between `<description>` and `<licenses>` like this:

```xml
    <name>...</name>
    <description>
        ...
    </description>

    <properties>
        all properties here
    </properties>

    <licenses>
        ...
    </licenses>
```

Regarding DCO: All you need to do is run

```
git commit --signoff
```

See [https://github.com/apps/dco](https://github.com/apps/dco).

Thanks again!

File: prometheus-metrics-tracer/prometheus-metrics-tracer-otel-agent/pom.xml [hunk 3]
```diff
@@ -32,6 +32,10 @@
     </developer>
   </developers>
 
+  <properties>
+    <automatic-module-name>io.prometheus.metrics.tracer.otel.agent</automatic-module-name>
```
Comment (fstab) [inline prometheus-metrics-tracer/prometheus-metrics-tracer-otel-agent/pom.xml:36]: The Java package name is `io.prometheus.metrics.tracer.otel_agent`.

File: pom.xml (hunk 1 above)
Comment (solomax) [reply pom.xml:69]: done

File: prometheus-metrics-tracer/prometheus-metrics-tracer-otel-agent/pom.xml (hunk 3 above)
Comment (solomax) [reply prometheus-metrics-tracer/prometheus-metrics-tracer-otel-agent/pom.xml:36]: done

**Comment by solomax** [comment]
> Thanks a lot @solomax!
> 
> I added a few minor comments. One more thing: We are trying to keep a similar structure across all `pom.xml` files. Please move the `<properties>` sections so that it's between `<description>` and `<licenses>` like this:
> 
> ```
>     <name>...</name>
>     <description>
>         ...
>     </description>
> 
>     <properties>
>         all properties here
>     </properties>
> 
>     <licenses>
>         ...
>     </licenses>
> ```
> 
> Regarding DCO: All you need to do is run
> 
> ```
> git commit --signoff
> ```
> 
> See https://github.com/apps/dco.
> 
> Thanks again!

This will require new PR :(

will close this one ...

**Pull request closed without merging**